src/__tests__/hooks/medium.useEventOperations.spec.tsx

```diff
@@ -0,0 +1,255 @@
+        category: '업무',
+        repeat: { type: 'none', interval: 0, endDate: '' },
+        notificationTime: 1,
+      } as EventForm;
+
+      // WHEN: 새로 정의된 데이터를 업데이트 했을때
+      act(() => queryEvent.current.createEvent.createEvent(mockCreateEvent));
+
+      // THEN: 새로 정의된 데이터를 불러와야 한다.
+      await waitFor(() => {
+        const eventsData = queryEvent.current.fetchEvents.events;
+        const expectedNewEvent = { id: String(MOCK_EVENTS.length + 1), ...mockCreateEvent };
+        expect(eventsData).toContainEqual(expectedNewEvent);
+      });
+    });
+  });
+
+  describe('delete event', () => {
+    it('존재하는 이벤트 삭제 시 에러없이 아이템이 삭제된다.', async () => {
+      // GIVEN: 데이터 불러오기
+      await waitFor(() =>
+        expect(queryEvent.current.fetchEvents.events).toHaveLength(MOCK_EVENTS.length)
+      );
+      MOCK_EVENTS.forEach((event) =>
+        expect(queryEvent.current.fetchEvents.events).toContainEqual(event)
+      );
+
+      // WHEN: 데이터를 삭제 했을 때
+      act(() => queryEvent.current.deleteEvent.deleteEvent('1'));
+
+      // THEN: 삭제된 데이터는 이벤트 목록에 존재하지 않는다.
+      await waitFor(() => {
+        const eventsData = queryEvent.current.fetchEvents.events;
+        expect(eventsData).toHaveLength(MOCK_EVENTS.length - 1);
+        expect(queryEvent.current.fetchEvents.events).not.toContainEqual({ ...MOCK_EVENTS[0] });
+      });
+    });
+  });
+
+  it("존재하지 않는 이벤트 수정 시 '일정 저장 실패'라는 토스트가 노출되며 에러 처리가 되어야 한다", async () => {
+    // GIVEN: 데이터 불러오기
+    await waitFor(() =>
+      expect(queryEvent.current.fetchEvents.events).toHaveLength(MOCK_EVENTS.length)
+    );
+    const NOT_FOUND_EVENT = {
+      id: 'test-mock-event-id',
+      title: 'test',
+      date: '',
+      startTime: '',
+      endTime: '',
+      description: '',
+      location: 'a',
+      category: 'aa',
+      repeat: { type: 'none', interval: 0, endDate: '' },
+      notificationTime: 60,
+    } as Event;
+
+    // WHEN: 존재하지 않는 이벤트 수정했을 때
+    act(() => queryEvent.current.updateEvent.updateEvent({ ...NOT_FOUND_EVENT }));
+    await waitFor(() => expect(toastCalls.length).toBeGreaterThan(0));
+
+    // THEN: 일정 저장 실패 토스트 노출, 에러 처리가 되어야 한다.
+    expect(toastCalls).toContainEqual({
+      title: '일정 저장 실패',
+      status: 'error',
+      duration: 3000,
+      isClosable: true,
+    });
+    expect(queryEvent.current.fetchEvents.events).not.toStrictEqual(NOT_FOUND_EVENT);
+  });
+});
+
+describe('api 에러 반환', () => {
+  let fetchEvents: {
+    current: ReturnType<typeof useFetchEvents>;
+  };
+
+  const toast = useToast();
+  const queryClient = createTestQueryClient(toast);
+  const wrapper = createWrapper(queryClient);
+
+  beforeEach(() => {
+    // GIVEN: 각 테스트마다 초기 상태가 반영된 핸들러 설정
+    toastCalls = [];
+    queryClient.clear();
+  });
+  afterEach(() => vi.restoreAllMocks());
+
+  it("이벤트 로딩 실패 시 '이벤트 로딩 실패'라는 텍스트와 함께 에러 토스트가 표시되어야 한다", async () => {
+    // GIVEN: fetch 함수 mocking
+    vi.spyOn(fetchJsonModule, 'fetchJson').mockRejectedValue(new Error('이벤트 로딩 실패'));
```

Comment:
p1;
msw의 server 오버라이딩을 활용하시면 될 듯 합니다.

```suggestion
    server.use(
    http.get('/api/events', () => {
      return new HttpResponse(null, { status: 500 });
    })
  );
```

Reply:
setupTest.ts 파일의 beforeEach에서 server.use(...setupMockHandlers([...initialEvents] as Event[]));
오버라이딩하는데 강제 에러를 내야해서 spyon썼습니다!